docs/canonicalk8s/charm/howto/install/install-lxd.md

```diff
@@ -8,122 +8,20 @@ require some extra steps. These are:
 - deploying to a container on a machine (i.e. when installing a bundle or using
   the 'to:' directive to install to an existing machine)
 
-The container running the charm, or more accurately, the LXD instance
-controlling the container, needs to have a particular configuration in order
-for the Kubernetes components to operate properly.
+LXD is about to remove the support for privileged containers and some
+Kubernetes services, such as the Cilium CNI, cannot run inside unprivileged
+containers.
 
+As such, we recommend using LXD virtual machines, which also ensure that the
+Kubernetes environment is well isolated.
 
-## Apply the {{product}} LXD profile
+We can use Juju constraints to request virtual machines to be used instead
+of containers and specify the amount of resources to allocate.
 
-On the machine running the 'localhost' cloud, we can determine the existing
-profiles by running the command:
+For example, we can pass the following constraints when deploying ``k8s``:
 
 ```
-lxc profile list
+juju deploy k8s --channel=$channel \
+  --constraints='cores=2 mem=4G root-disk=40G virt-type=virtual-machine'
 ```
```

Comment:
There seems to be no special steps for deploying with LXD VMs besides adding extra resources. Are there any steps for the first scenario listed of deploying to the 'localhost' cloud? If not I think this page can be completely reworked. Right now it says you can do this then it says not you can't

Reply:
> Are there any steps for the first scenario listed of deploying to the 'localhost' cloud?

The 'localhost' built-in juju cloud uses lxd so the same restrictions apply:

```
Cloud      Regions  Default    Type  Credentials  Source    Description
localhost  1        localhost  lxd   1            built-in  LXD Container Hypervisor
```

>  Right now it says you can do this then it says not you can't

Indeed, right now it says ``deploying to a container on a machine``, I'll need to update this.

Reply:
This is clearer thank you :smile_cat: